app/services/uphold/refresher.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Uphold
+  class Refresher
+    def self.build
+      new(impl_refresher: Uphold::Models::Authorization.new)
+    end
+
+    def initialize(impl_refresher:)
+      @impl_refresher = impl_refresher
+    end
+
+    # Makes a request to the Uphold API to refresh the current access_token
+    def call(uphold_connection:)
+      # Ensure we have an refresh_token.
+      refresh_token = uphold_connection.refresh_token
+
+      return if refresh_token.blank?
+      return if uphold_connection.authorization_expires_at&. > Time.zone.now
+
+      authorization = @impl_refresher.refresh_authorization(uphold_connection)
+
+      # add expiration time
+      authorization_hash = JSON.parse(authorization)
+
+      authorization_hash["expiration_date"] = authorization_hash["expires_in"].to_i.seconds.from_now.to_s
```

Comment:
Would you make `expiration_date` a constant? Also, would you call it `expiration_time`?

Reply:
I can rename it, but it's a value we receive from uphold, so it should stay a var.

Reply:
Renamed.